### PR TITLE
BTAT-4738: Added dereg confirmation notification pref logic for agent and non-agent

### PR DESCRIPTION
**Pull request by davidmthornton**

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -147,9 +147,12 @@ checkYourAnswers.confirm = Confirm and deregister
...
 deregistrationConfirmation.heading = What happens next
-deregistrationConfirmation.text = If the business can deregister from VAT, we will send a confirmation letter. This usually takes 10 working days.
-deregistrationConfirmation.link = Check your contact details
-deregistrationConfirmation.link.text = are up to date.
+deregistrationConfirmation.textNonAgent = We will send you an email within 2 working days with an update, followed by a letter to your principal place of business. You can also go to your HMRC secure messages to find out if your request has been accepted.
+deregistrationConfirmation.text2NonAgent = Ensure your contact details are up to date.
+deregistrationConfirmation.textAgentPrefYes = We will send an email to {0} within 2 working days telling you whether or not the request has been accepted.
+deregistrationConfirmation.text2AgentPrefYes = We will also contact your client with an update.
+deregistrationConfirmation.textAgentPrefNo = We will send you a confirmation letter to your principal place of business within 12 working days telling you whether or not the request has been accepted.
```
Comment (TomDudley92) [inline conf/messages]: According to the images on the JIRA story this should be 10 days not 12.

File: app/views/deregistrationConfirmation.scala.html [hunk 2]
```diff
@@ -27,16 +29,26 @@ <h1 class="heading-xlarge">@messages("deregistrationConfirmation.title")</h1>
 
 <h2 class="heading-medium">@messages("deregistrationConfirmation.heading")</h2>
 
-<p>@messages("deregistrationConfirmation.text")</p>
+@{
+    if(user.isAgent){
+        user.session.get(SessionKeys.verifiedAgentEmail) match {
+            case Some(verifiedEmail) =>
+                <p>{messages("deregistrationConfirmation.textAgentPrefYes", verifiedEmail)}</p>
```
Comment (TomDudley92) [inline app/views/deregistrationConfirmation.scala.html]: The verified email should be in bold as per the prototype.

File: test/views/DeregistrationConfirmationSpec.scala [hunk 3]
```diff
@@ -20,46 +20,103 @@ import assets.messages.{CommonMessages, DeregistrationConfirmationMessages}
...
+
+      "have the correct first paragraph" in {
+        elementText(Selectors.text) shouldBe DeregistrationConfirmationMessages.textAgentPrefYes
+      }
+
+      "have the correct text for the second paragraph" in {
+        elementText(Selectors.text2) shouldBe DeregistrationConfirmationMessages.text2AgentPrefYes
+      }
+
+      s"have the correct continue button text and url" in {
```
Comment (TomDudley92) [inline test/views/DeregistrationConfirmationSpec.scala]: This isn't testing the url

File: test/views/DeregistrationConfirmationSpec.scala [hunk 4]
```diff
@@ -20,46 +20,103 @@ import assets.messages.{CommonMessages, DeregistrationConfirmationMessages}
...
+
+      "have the correct first paragraph" in {
+        elementText(Selectors.text) shouldBe DeregistrationConfirmationMessages.textAgentPrefNo
+      }
+
+      "have the correct text for the second paragraph" in {
+        elementText(Selectors.text2) shouldBe DeregistrationConfirmationMessages.text2AgentPrefNo
+      }
+
+      s"have the correct continue button text and url" in {
```
Comment (TomDudley92) [inline test/views/DeregistrationConfirmationSpec.scala]: This isn't testing the url

File: conf/messages (hunk 1 above)
Comment (davidmthornton) [reply conf/messages]: Yeah Barry said it is 12.

File: app/views/deregistrationConfirmation.scala.html [hunk 5]
```diff
@@ -27,16 +29,28 @@ <h1 class="heading-xlarge">@messages("deregistrationConfirmation.title")</h1>
 
 <h2 class="heading-medium">@messages("deregistrationConfirmation.heading")</h2>
 
-<p>@messages("deregistrationConfirmation.text")</p>
+@{
+    if(user.isAgent){
+        user.session.get(SessionKeys.verifiedAgentEmail) match {
+            case Some(verifiedEmail) =>
+                <p>{messages("deregistrationConfirmation.textAgentPrefYesP1")}
+                    <strong>{verifiedEmail}</strong>
```
Comment (TomDudley92) [inline app/views/deregistrationConfirmation.scala.html]: The `<strong>` tag hasn't actually made the text bold. Try adding `class="bold"`

File: test/views/DeregistrationConfirmationSpec.scala [hunk 6]
```diff
@@ -20,46 +20,106 @@ import assets.messages.{CommonMessages, DeregistrationConfirmationMessages}
...
+        elementText(Selectors.text) shouldBe DeregistrationConfirmationMessages.textNonAgent
+      }
+
+      "have the correct text for the second paragraph" in {
+        elementText(Selectors.text2) shouldBe DeregistrationConfirmationMessages.text2NonAgent
+      }
 
-    "have the correct first paragraph" in {
-      elementText(Selectors.text) shouldBe DeregistrationConfirmationMessages.text
+      s"have the correct continue button text and url" in {
```
Comment (TomDudley92) [inline test/views/DeregistrationConfirmationSpec.scala]: Sorry can these tests be split out please

File: test/views/DeregistrationConfirmationSpec.scala [hunk 7]
```diff
@@ -20,46 +20,106 @@ import assets.messages.{CommonMessages, DeregistrationConfirmationMessages}
...
+        elementText(Selectors.text) shouldBe DeregistrationConfirmationMessages.textAgentPrefYes
+      }
+
+      "have the correct text for the second paragraph" in {
+        elementText(Selectors.text2) shouldBe DeregistrationConfirmationMessages.text2AgentPrefYes
+      }
+
+      s"have the correct continue button text and url" in {
+        element(Selectors.button).attr("href") shouldBe mockConfig.manageVatSubscriptionFrontendUrl
+        elementText(Selectors.button) shouldBe CommonMessages.finish
```
Comment (TomDudley92) [inline test/views/DeregistrationConfirmationSpec.scala]: Can these tests be split out please

**Pull request merged**